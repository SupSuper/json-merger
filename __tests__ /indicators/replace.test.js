const helpers = require("../../__helpers__");

describe("when merging two objects and a source property has a $replace indicator", function () {

    test("it should not merge the source property but replace it", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$replace": true,
                "bb": "replaced"
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("and a $value indicator it should not merge the source property but replace it with the $value result", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$replace": true,
                "$value": {
                    "bb": "replaced"
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("and a $ref indicator it should not merge the source property but replace it with the $ref result", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$replace": true,
                "$ref": "/b"
            },
            "b": {
                "bb": "replaced"
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("and a $merge indicator it should not merge the source property but replace it with the $merge result", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$replace": true,
                "$merge": {
                    "source": "notImportant",
                    "with": {
                        "bb": "replaced"
                    }
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });
});
