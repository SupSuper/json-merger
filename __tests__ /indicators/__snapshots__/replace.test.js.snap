// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $replace indicator and a $merge indicator it should not merge the source property but replace it with the $merge result 1`] = `
"{
	\\"a\\": {
		\\"bb\\": \\"replaced\\"
	}
}"
`;

exports[`when merging two objects and a source property has a $replace indicator and a $ref indicator it should not merge the source property but replace it with the $ref result 1`] = `
"{
	\\"a\\": {
		\\"bb\\": \\"replaced\\"
	},
	\\"b\\": {
		\\"bb\\": \\"replaced\\"
	}
}"
`;

exports[`when merging two objects and a source property has a $replace indicator and a $value indicator it should not merge the source property but replace it with the $value result 1`] = `
"{
	\\"a\\": {
		\\"bb\\": \\"replaced\\"
	}
}"
`;

exports[`when merging two objects and a source property has a $replace indicator it should not merge the source property but replace it 1`] = `
"{
	\\"a\\": {
		\\"bb\\": \\"replaced\\"
	}
}"
`;
